stacks-common/src/codec/mod.rs

```diff
@@ -183,6 +193,66 @@ pub fn read_next_exact<R: Read, T: StacksMessageCodec + Sized>(
     read_next_vec::<T, R>(fd, num_items, 0)
 }
 
+pub fn read_next_with_epoch<T: DeserializeWithEpoch, R: Read>(
```

Comment:
I don't think this method or `read_next_vec_with_epoch` are necessary or desirable.  Codec implementations that need to be epoch-aware should directly inspect the data they decode and decide how to proceed from there.